analyses/snv-callers/compare_snv_callers_plots-tcga.Rmd

```diff
@@ -124,141 +115,165 @@ all_caller_df <- strelka %>%
                    suffix = c("_strelka", "_mutect")) %>%
   dplyr::full_join(lancet, 
                    by = join_cols) %>% 
-  dplyr::rename(VAF_lancet = VAF)
+  dplyr::rename(VAF_lancet = VAF) %>% 
+  # We'll use this to keep track of the original rows/mutations
+  tibble::rowid_to_column("index")
 
 # Take a peek at what this looks like
 head(all_caller_df)
 ```
 
+## Get consensus numbers
+
+Make a data.frame with only the VAFs and turn into detected or not. 
+
 ```{r}
-# Set up as data.frame with each column a caller's VAF
-vaf_df <- all_caller_df %>% 
-  dplyr::select("VAF_lancet", "VAF_strelka", "VAF_mutect",
-                "Variant_Classification") %>%
-  dplyr::mutate(VAF_consensus = VAF_strelka)     
- 
-# Determine which mutations are not part of the consensus
-consensus_index <- vaf_df %>%
-  dplyr::select(-c(VAF_consensus, Variant_Classification)) %>%
-  rowSums(is.na(.)) > 0
-
-# Make non consensus VAF into an NA
-vaf_df$VAF_consensus[consensus_index] <- NA
-
-# Make this long form for plotting
-long_vaf_df <- vaf_df %>% 
-  # This index needs to be a factor so tapply step in the next chunk works more seamlessly
-  dplyr::mutate(index = factor(1:nrow(.))) %>% 
-  tidyr::gather(key = "caller", value = "vaf", -index, -Variant_Classification) %>% 
-  dplyr::mutate(caller = gsub("VAF_", "", caller)) %>% 
-  dplyr::filter(!is.na(vaf))
+detect_mat <- all_caller_df %>% 
+ # Bring over VAF columns and the index
+ dplyr::select(dplyr::starts_with("VAF_")) %>% 
+ as.matrix()
+
+# Store the indices as dimnames
+dimnames(detect_mat)[[1]] <- all_caller_df$index
+
+# Turn into detected or not
+detect_mat <- !is.na(detect_mat)
 ```
 
-Get median vaf for each mutation with each caller combination. 
+Count up how many callers call each mutation but ignore VarDict. 
```

Comment:
It looks like this table already ignores vardict from the select statements above. ( I guess this is probably tcga-specific)